llvm/test/CodeGen/AMDGPU/lower-work-group-id-intrinsics-hsa.ll

```diff
@@ -0,0 +1,147 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+; RUN: llc -mtriple=amdgcn-amd-hsa -mcpu=gfx900 -mattr=+architected-sgprs --verify-machineinstrs < %s | FileCheck -check-prefix=GFX9-SDAG %s
```

Comment:
I don't think we have standalone tests that cover workgroup IDs so far.
Can you add a run line for gfx900 (or any arch before gfx12) without -mattr=archited-sgpr?
It makes more sense as this patch refactors the existing code that handles the workgroup IDs too.
Also, why no RUN line for gfx1200?

Reply:
> I don't think we have standalone tests that cover workgroup IDs so far.

What about `test/CodeGen/AMDGPU/llvm.amdgcn.workgroup.id.ll`?

> Also, why no RUN line for gfx1200?

We have not enabled FeatureArchitectedSGPRs for GFX12 yet (I was going to do that in a later patch) so it would not test anything interesting yet.

Reply:
> Can you add a run line for gfx900 (or any arch before gfx12) without -mattr=archited-sgpr?

Done.

Reply:
> We have not enabled FeatureArchitectedSGPRs for GFX12 yet (I was going to do that in a later patch)

Actually I guess enabling it for GFX12 is orthogonal, so I have done that here: #79160